src/python/espressomd/visualizationOpenGL.pyx

```diff
@@ -327,15 +326,17 @@ class openGLLive:
             ptype = self.particles['types'][pid]
             ext_f = self.particles['ext_forces'][pid]
 
+            radius = 1
             # Size: Lennard Jones Sigma,
-            if self.specs['particle_sizes'] == 'auto':
-                lj_sig = self.system.non_bonded_inter[ptype, ptype].lennard_jones.get_params()[
-                    'sigma']
+            if callable(self.specs['particle_sizes']):
+                radius = self.specs['particle_sizes'](ptype)
+            elif self.specs['particle_sizes'] == 'auto':
+                lj_sig = self.system.non_bonded_inter[ptype, ptype].lennard_jones.get_params()['sigma']
                 radius = lj_sig / 2.0
                 if radius == 0:
                     radius = self.sizeByType(ptype)
 
-            elif self.specs['particle_sizes'] == 'type':
+            elif isinstance(self.specs['particle_sizes'], list):
```

Comment:
This only applies to `list`, but you should accept anything that has a subscript operator (e.g. a Numpy array). Please see #905 for how to do that with a try-except. Also, I think the line below should be something like 
```
radius = self.specs['particle_sizes'][ptype]
```